shortestBridge.js

```diff
@@ -0,0 +1,187 @@
+*/
+
+const { Queue } = require('../../../utils/queue');
+
+const LAND = 1;
+
+const directions = [
+    [0, 1],
+    [1, 0],
+    [0, -1],
+    [-1, 0],
+];
+
+// 0 1 0
+// 0 0 0
+// 0 0 1
+
+const isBound = (row, col, grid) => {
+    const numRows = grid.length;
+    const numCols = grid[0].length;
+
+    const isRowBound = row >= 0 && row < numRows;
+    const isColBound = col >= 0 && col < numCols;
+
+    return isRowBound && isColBound;
+};
+
+const getRowColString = ({ row, col }) => `${row}, ${col}`;
+
+const getChildren = (row, col, grid) => {
+    const children = [];
+
+    for (const direction of directions) {
+        const [rowDir, colDir] = direction;
+
+        const newRow = row + rowDir;
+        const newCol = col + colDir;
+
+        if (!isBound(newRow, newCol, grid)) continue;
+
+        children.push({ row: newRow, col: newCol });
+    }
+
+    return children;
+};
+
+const findFirstIslandCoordinates = (startRow, startCol, grid) => {
+    const visited = new Set();
+    const queue = new Queue();
+
+    const rowColString = getRowColString({ row: startRow, col: startCol });
+    visited.add(rowColString);
+
+    queue.enqueue({ row: startRow, col: startCol });
+
+    const firstIslandCoordinates = [];
+    while (queue.size() > 0) {
+        // Remove node
+        const { row, col } = queue.dequeue();
+
+        // Process node
+        if (grid[row][col] !== LAND) continue;
+        firstIslandCoordinates.push({ row, col });
+
+        // Get children
+        const children = getChildren(row, col, grid);
+        for (const child of children) {
+            const rowColString = getRowColString(child);
+            if (visited.has(rowColString)) continue;
+            visited.add(rowColString);
+
+            queue.enqueue({ ...child });
+        }
+    }
+
+    return firstIslandCoordinates;
+};
+
+const getFirstIslandCoordinates = (grid) => {
+    const numRows = grid.length;
+    const numCols = grid[0].length;
+
+    let firstIslandCoordinates = null;
+
+    for (let row = 0; row < numRows; row++) {
+        for (let col = 0; col < numCols; col++) {
+            if (grid[row][col] !== LAND) continue;
+
+            firstIslandCoordinates = findFirstIslandCoordinates(row, col, grid);
+
+            return firstIslandCoordinates;
+        }
+    }
+};
+
+const getShortestBridge = (grid, firstIslandCoordinates) => {
+    const visited = new Set();
+    const queue = new Queue();
+
+    for (const firstIslandCoordinate of firstIslandCoordinates) {
+        visited.add(getRowColString(firstIslandCoordinate));
+        queue.enqueue({ ...firstIslandCoordinate, path: 0 });
+    }
+
+    while (queue.size() > 0) {
+        // Remove node
+        const { row, col, path } = queue.dequeue();
+
+        // Process node
+        const isIslandVisited = visited.has(getRowColString({ row, col }));
+        const terrain = grid[row][col];
+
+        if (!isIslandVisited && terrain === LAND) {
```

Comment:
What does this condition represent? Might be good to create a well-named boolean variable here.

Reply:
This condition checks if we've came across an unvisited location for the first time and if it's an island. If those two conditions are met, we've found the second island path and we're able to return the path so far.